toDoApp/src/navigation/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import { createStackNavigator } from '@react-navigation/stack'
+import { Home } from '../screens/Home'
+import { styles } from './styles'
+import { Color } from '../styles/Pallete'
+import { Pressable, Text } from 'react-native'
+import { NewTask } from '../screens/NewTask'
+import { Details } from '../screens/Details'
+import { homeProps, newTaskProps, RootStackParamList } from './types'
+import { constants, routes } from '../common/constants'
+
+export const AppContainer = () => {
+  const Stack = createStackNavigator<RootStackParamList>()
+
+  const homeHeader = ({ navigation }: homeProps) => ({
+    title: constants.homeTitle,
+    headerStyle: styles.header,
+    headerTintColor: Color.almostWhite,
+    headerRight: () => (
+      <Pressable
+        style={styles.button}
+        onPress={() => navigation.navigate(routes.newTask)}>
+        <Text style={styles.addButton}>+</Text>
+      </Pressable>
+    ),
+  })
```

Comment:
This shouldn't be here, what happens if you need something from the scene when you need to tap the header?
THe best approach its to maintain concerns, what belongs to the scene should go in the scene.